src/main/java/mate/academy/intro/repository/cartitem/CartItemRepository.java

```diff
@@ -3,9 +3,15 @@
 import java.util.Optional;
 import mate.academy.intro.model.CartItem;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
 
 public interface CartItemRepository extends JpaRepository<CartItem, Long> {
     Optional<CartItem> findCartItemByIdAndShoppingCartId(
             Long id,
             Long shoppingCardId);
+
+    @Modifying
+    @Query("UPDATE CartItem c SET c.isDeleted = true WHERE c.shoppingCart.id = :userId")
+    void deleteByShoppingCartAndId(Long userId);

```

Comment:
Consider whether the cart item really needs to be a soft deleted entity. How quickly will the table fill up with data? How will this affect performance? Do you really need to keep this entity for a long time (same goes for order items)